TombEngine/Scripting/Internal/TEN/Objects/Lara/LaraObject.cpp

```diff
@@ -98,6 +99,81 @@ int LaraObject::GetAir() const
 	return lara->Air;
 }
 
+//Lara will undraw her weapons if they are drawn, throw away flare if she helds one in her hand.
+// @function LaraObject:UndrawWeapons
+void LaraObject::UndrawWeapons()
+{
+	auto* lara = GetLaraInfo(m_item);
+
+	if (lara->Control.HandStatus != HandStatus::Free ||
+		lara->Control.Weapon.GunType == LaraWeaponType::Flare)
+	{
+		lara->Control.HandStatus = HandStatus::WeaponUndraw;
+	}
+}
+
+//Lara will throw away the torch if she helds one in her hand.
+// @function LaraObject:ThrowAwayTorch
+void LaraObject::ThrowAwayTorch()
+{
+	auto* lara = GetLaraInfo(m_item);
+
+	if (lara->Control.Weapon.GunType == LaraWeaponType::Torch)
+	{
+		Lara.Torch.State = TorchState::Dropping;
+	}
+}
+
+// @function LaraObject:GetLaraHandStatus
+// @treturn 0=HandsFree, 1=Busy(climbing,etc), 2=WeaponDraw, 3=WeaponUndraw, 4=WeaponInHand.
+HandStatus LaraObject::GetLaraHandStatus() const
+{
+	auto* lara = GetLaraInfo(m_item);
+	return  HandStatus{ lara->Control.HandStatus };
+}
+
+// @function LaraObject:GetLaraWeaponType
+// @treturn 0=None, 1=Pistols, 2=Revolver, 3=Uzi, 4=Shotgun, 5=HK, 6=Crossbow, 7=Flare, 8=Torch, 9=GrenadeLauncher, 10=Harpoon, 11=RocketLauncher.
+LaraWeaponType LaraObject::GetLaraWeaponType() const
+{
+	auto* lara = GetLaraInfo(m_item);
+	return LaraWeaponType{ lara->Control.Weapon.GunType };
+}
+
+//Set Lara Weapon type
+// @function LaraObject:SetLaraWeaponType
+// 0=None, 1=Pistols, 2=Revolver, 3=Uzi, 4=Shotgun, 5=HK, 6=Crossbow, 7=Flare, 8=Torch, 9=GrenadeLauncher, 10=Harpoon, 11=RocketLauncher.
+void LaraObject::SetLaraWeaponType(LaraWeaponType weaponType, bool activate)
```

Comment:
Same as above. Also, comments are formatted in a wrong way again.

Reply:
changed to SetWeaponType. Changed formatting